angular/projects/admin-nrpti/src/app/mines/mines-add-edit/mines-add-edit.component.ts

```diff
@@ -16,105 +17,197 @@ export class MinesAddEditComponent implements OnInit, OnDestroy {
+      links.push({
+        title: linkFormGroup.title,
+        url: linkFormGroup.url
+      });
+    });
+
+    return links;
+  }
+
+  /**
+   * Tracks type select formcontrol changes.
+   *
+   * @param {*} optionA
+   * @param {*} optionB
+   * @returns {boolean}
+   * @memberof MinesAddEditComponent
+   */
   compareTypeSelection(optionA: any, optionB: any): boolean {
     return optionA === optionB;
   }
 
-  navigateBack() {
-    if (this.isEditing) {
-      this.router.navigate(['mines', this.record._id, 'detail']);
-    } else {
-      this.router.navigate(['mines']);
+  cancel() {
+    const shouldCancel = confirm(
+      'Leaving this page will discard unsaved changes. Are you sure you would like to continue?'
+    );
+    if (shouldCancel) {
+      this.router.navigate(['mines', this.mine._id, 'detail']);
     }
   }
 
-  async submit() {
-    this.loadingScreenService.setLoadingState(true, 'main');
+  /**
+   * Toggle the publish formcontrol.
+   *
+   * @param {*} event
+   * @memberof MinesAddEditComponent
+   */
+  togglePublish(event) {
+    this.myForm.controls.publish.setValue(event.checked);
+
+    this._changeDetectionRef.detectChanges();
+  }
+
+  /**
+   * Parses the form data into a mine object.
+   *
+   * @returns mine object
+   * @memberof MinesAddEditComponent
+   */
+  buildMineObject() {
     const mineItem = {};
-    mineItem['_id']                  = this.record._id;
-    mineItem['_schemaName']          = this.record._schemaName;
-    mineItem['_sourceRefId']         = this.record._sourceRefId;
-    mineItem['name']                 = this.record.name;
-    mineItem['permitNumbers']        = this.record.permitNumbers;
-    mineItem['status']               = this.record.status;
-    mineItem['type']                 = this.myForm.controls.type.value;
-    mineItem['commodities']          = this.record.commodities;
-    mineItem['tailingsImpoundments'] = this.record.tailingsImpoundments;
-    mineItem['region']               = this.record.region;
-    mineItem['location']             = this.record.location;
-    mineItem['permittee']            = this.record.permitee;
-    mineItem['summary']              = this.myForm.controls.summary.value;
-    mineItem['description']          = this.myForm.controls.description.value;
-    mineItem['links']                = this.myForm.controls.links.value.split(',');
 
-    this.loadingScreenService.setLoadingState(false, 'main');
+    mineItem['_id'] = this.mine._id;
 
-    if (!this.isEditing) {
-      // Add the mine item.
-      // Are we manually adding mines or is this import only?
-      this.router.navigate(['mines', this.record._id, 'detail']);
-    } else {
-      // Update the mine item.
-      this.factoryService.editMine(mineItem).subscribe(async res => {
-        this.router.navigate(['mines', this.record._id, 'detail']);
-      });
+    this.myForm.get('type').dirty && (mineItem['type'] = this.myForm.get('type').value);
```

Comment:
Isn't `factoryService.editMine()` performing a `PUT` request? By only selecting the fields that have been altered it looks like this object is being prepared for a `PATCH` request.

Reply:
All of our add-edits currently do this same dirty checking + PUT request.  Since they all use a single common controller, it probably needs to stay this way for now, and then if they should all in fact be PATCHES, that could be its own ticket.